docs/INSTALL.md

```diff
@@ -0,0 +1,518 @@
+
+### Redirect all unencrypted traffic to HTTPS
+
+To redirect all HTTP traffic to HTTPS you should issue a permanent redirect using the 301 status code:
+
+```
+<VirtualHost *:80>
+   ServerName yourservername.org
+   Redirect permanent / https://yourservername.org/
+</VirtualHost>
+```
+
+### Enable HTTP Strict Transport Security
+
+By adding the following to your Apache SSL VirtualHost configuration you instruct browsers not to allow any connection to your ILIAS instance using HTTP and prevent visitors from bypassing invalid certificate warnings.
+
+```
+<IfModule mod_headers.c>
+    Header always set Strict-Transport-Security "max-age=15768000; includeSubDomains; preload"
+</IfModule>
+```
+
+**Warning:** Before activating the configuration above make sure that you are absolutely sure that you have a good workflow for maintaining your SSL settings (including certificate renewals) as you will not be able to disable HTTPS access to your site for up to 6 months.
+
+### Proper SSL configuration
+
+The default SSL ciphers used by web servers are often not state-of-the-art, so you should consider to choose your own settings. Which settings should be used depends completely on your environment. Therefore giving a generic recommendation is not really possible.
+
+We recommend to use the [Mozilla SSL Configuration Generator](https://mozilla.github.io/server-side-tls/ssl-config-generator/) to generate a suitable configuration and the [Qualys SSL Labs Tests](https://www.ssllabs.com/ssltest/) or the [High-Tech Bridge SSL Server Test](https://www.htbridge.com/ssl/) to check your settings.
+
+## Serve security related Headers
+
+To improve the security of your ILIAS users you should set at least the following Headers:
+
+  * X-Content-Type-Options: nosniff
+  * X-XSS-Protection: 1; mode=block
+  * X-Frame-Options: SAMEORIGIN
+
+## Report security issuses
+
+If you think you found an security related issue in ILIAS please refer to http://www.ilias.de/docu/goto_docu_wiki_5307.html#ilPageTocA213 for reporting it.
+
+For Apache on Debian systems you can turn those headers on by editing ```/etc/apache2/conf-enabled/security.conf```, make sure ```mod_headers``` and ```mod_env``` are enabled.
+
+For Nginx you can simply add for example ```add_header X-Frame-Options "SAMEORIGIN";``` in your ```server``` configuration.
+
+# Customizing ILIAS
+
+If you need to customize your ILIAS installation avoid under all circumstances to edit the core files, otherwise you will not be able to update your installation in a timely manner (e.g. due to security fixes). 
+
+You can find proper ways to customize ILIAS in the [ILIAS Development Guide](http://www.ilias.de/docu/goto_docu_pg_29964_42.html):
+
+  * [Plugins and Plugin Slots - ILIAS Development Guide](http://www.ilias.de/docu/goto.php?target=st_27029)
+  * [Styles and Templates - ILIAS Development Guide](http://www.ilias.de/docu/goto_docu_pg_210_42.html)
```

Comment:
@Amstutz I've replaced the "Styles and Templates" link with the one you provided.